### ISSUE TITLE
aws-vpc-cni chart duplicates toleration key in DaemonSet

### ISSUE THREAD

**Issue by Bourne-ID** [aws/amazon-vpc-cni-k8s#2344]
**What happened**:
When generating the aws-vpc-cni helm chart, if a toleration is defined in the values.yaml file then the key of tolerations is duplicated causing an invalid YAML syntax. Whilst kubernetes overrides this by applying the last toleration defined in the file, any YAML sanity checking tools should highlight the duplicate key, for example Kustomize will error with 
```
Error: map[string]interface {}(nil): yaml: unmarshal errors:
  line 182: mapping key "tolerations" already defined at line 47
```

**What you expected to happen**:
Only one toleration is defined when tolerations are provided in values.yaml, overriding the default toleration of `- operator: Exists`

**How to reproduce it (as minimally and precisely as possible)**:
* Define a toleration in values.yaml, for example
```yaml
tolerations:
  - effect: NoSchedule
    key: nvidia.com/gpu
    operator: Equal
    value: present
```

* Generate the helm file `helm template -f values.yaml .` 
* Observe 2 toleration keys defined for the daemonset, one for `- operator: Exists` and one as above. 

It's worth noting if applied to Kubernetes, it will apply the last toleration defined - it's the generated YAML syntax which is the issue. 

**Anything else we need to know?**:
If accepted this is going to be an interesting one to solve. One option would be to move the `exists` tolerance into the default values.yaml, thereby shifting the default setting into the values file. However this could be a breaking change for anyone upgrading with a copy of the old values.yaml file. 

To keep the same behavior, logic such as `{{ if eq (len .Values.tolerations) 0 }}` could be added so if the list is empty the operator:exists is added. However this feels like assumptions are being made in the template opposed to being explicit defaults in the values.yaml file. 

Overall this is an edge case: Taking a use case example; I want to utilise unique settings for certain large nodes to help reduce the number of reserved IPs by utilising [WARM_ENI_TARGET, WARM_IP_TARGET and MINIMUM_IP_TARGET](https://github.com/aws/amazon-vpc-cni-k8s/blob/master/docs/eni-and-ip-target.md).

Whilst there are other ways to reduce or work around this issue (kustomize patching) it did highlight an issue with this helm chart. Happy to have a discussion for a path forwards and produce a PR based on the notes above. 

**Environment**:
Helm Version: version.BuildInfo{Version:"v3.11.1", GitCommit:"293b50c65d4d56187cd4e2f390f0ada46b4c4737", GitTreeState:"clean", GoVersion:"go1.19.5"

Thanks all, have a good day!

**Comment by Bourne-ID** [comment]
Added a PR for what I could classify as a patch-level bump on the chart - keeping with the same logic IE no tolerations defined in values.yaml and it'll add the toleration for operator: exists. If the decision is to go down moving the default to values.yaml then this would be an easy change and my exiting PR can be edited or closed.

**Comment by jdn5126** [comment]
Thanks for filing this @Bourne-ID ! I think the better solution is to move the existing `tolerations` block into the default `values.yaml`. Using helm to install new templates and an old version of `values.yaml` file is not a case we need to concern ourselves with, as we do not support this type of mixing and matching.

We appreciate you filing a PR as well! Let's go with the default `values.yaml` approach and a minor version bump.

**Comment by Bourne-ID** [comment]
Sounds good, will update the PR this evening.

**Comment by Bourne-ID** [comment]
PR has been updated based on feedback @jdn5126. Happy for you to amend for minor release, and let me know if you would like me to add anything else to this PR. Thanks!

**Comment by jdn5126** [comment]
Closing as fixed by #2345 . This chart update will ship with the next release
